gobblin-modules/gobblin-sql/src/test/java/org/apache/gobblin/writer/MySqlBufferedInserterTest.java

```diff
@@ -63,7 +93,7 @@ public void testMySqlBufferedInsert() throws SQLException {
     }
     inserter.flush();
 
-    verify(conn, times(2)).prepareStatement(anyString());
+    verify(conn, times(2)).prepareStatement(matches("REPLACE INTO .*"));
```

Comment:
nice--clear and direct.  could we do similar for the `MySqlWriterCommands`?  (does it have a unit test yet for you to build on?)

Reply:
The MySQL writer commands does not have a similar test, which is why I had modified the `JdbcPublisherTest` initially. There is a `JdbcWriterCommandsTest` that mocks a `preparedStatement` to submit and `execute` but it does not actually test insertions into the database. It only tries to read from a mock database (see `testMySqlDateTypeRetrieval`) without testing the other writer commands.

Reply:
ideally we'd have one, but pragmatically I understand the imperative to ship these changes, so up to you whether we leave this as tech debt IoU to return to